api/webhook/taskcluster.go

```diff
@@ -88,8 +88,34 @@ type statusEventPayload struct {
 	Branches  []branchInfo `json:"branches"`
 }
 
+// GetBranch returns the first branch that has the revision of the commit
+// as its HEAD state.
+func (s statusEventPayload) GetBranch() string {
+	// Only process the master branch.
+	for _, branch := range s.Branches {
+		if branch.Commit.Sha == s.Sha {
+			return branch.Name
+		}
+	}
+	return ""
+}
+
+func (s statusEventPayload) IsSuccess() bool {
+	return s.State == "success"
+}
+
+func (s statusEventPayload) IsTaskcluster() bool {
+	return strings.HasPrefix(s.Context, "Taskcluster")
+}
+
 type branchInfo struct {
-	Name string `json:"name"`
+	Name   string     `json:"name"`
+	Commit commitInfo `json:"commit"`
+}
+
+type commitInfo struct {
+	Sha string `json:"sha"`
+	URL string `json:"url"`
```

Comment:
Do we need the URL of a commit? If not, let's remove this field. The API payload from both GitHub and Taskcluster are big and complicated, so I like to include only the necessary fields in various structs, which simplifies the struct definition and avoids distraction.

Reply:
Done.